tests/e2e/multicluster/multicluster_primaryremote_test.go

```diff
@@ -0,0 +1,353 @@
+						Expect(samplePodsRemote.Items).ToNot(BeEmpty(), "No pods found in bookinfo namespace")
+
+						for _, pod := range samplePodsRemote.Items {
+							Eventually(common.GetObject).
+								WithArguments(ctx, clRemote, kube.Key(pod.Name, "sample"), &corev1.Pod{}).
+								Should(HaveCondition(corev1.PodReady, metav1.ConditionTrue), "Pod is not Ready on Remote; unexpected Condition")
+						}
+						Success("Sample app is created in both clusters and Running")
+					})
+
+					It("can access the sample app from both clusters", func(ctx SpecContext) {
+						sleepPodNamePrimary, err := common.GetPodNameByLabel(ctx, clPrimary, "sample", "app", "sleep")
+						Expect(sleepPodNamePrimary).NotTo(BeEmpty(), "Sleep pod not found on Primary Cluster")
+						Expect(err).NotTo(HaveOccurred(), "Error getting sleep pod name on Primary Cluster")
+
+						sleepPodNameRemote, err := common.GetPodNameByLabel(ctx, clRemote, "sample", "app", "sleep")
+						Expect(sleepPodNameRemote).NotTo(BeEmpty(), "Sleep pod not found on Remote Cluster")
+						Expect(err).NotTo(HaveOccurred(), "Error getting sleep pod name on Remote Cluster")
+
+						// Run the curl command from the sleep pod in the Remote Cluster and get response list to validate that we get responses from both clusters
+						remoteResponses := strings.Join(getListCurlResponses(sleepPodNameRemote, kubeconfig2), "\n")
+						Expect(remoteResponses).To(ContainSubstring("Hello version: v1"), "Responses from Remote Cluster are not the expected")
+						Expect(remoteResponses).To(ContainSubstring("Hello version: v2"), "Responses from Remote Cluster are not the expected")
+
+						// Run the curl command from the sleep pod in the Primary Cluster and get response list to validate that we get responses from both clusters
+						primaryResponses := strings.Join(getListCurlResponses(sleepPodNamePrimary, kubeconfig), "\n")
+						Expect(primaryResponses).To(ContainSubstring("Hello version: v1"), "Responses from Primary Cluster are not the expected")
+						Expect(primaryResponses).To(ContainSubstring("Hello version: v2"), "Responses from Primary Cluster are not the expected")
+						Success("Sample app is accessible from both clusters")
+					})
+				})
+
+				When("sample apps are deleted in both clusters", func() {
+					BeforeAll(func(ctx SpecContext) {
+						// Delete the entire sample namespace in both clusters
+						Expect(kubectl.DeleteNamespace("sample", kubeconfig)).To(Succeed(), "Namespace failed to be deleted on Primary Cluster")
+						Expect(kubectl.DeleteNamespace("sample", kubeconfig2)).To(Succeed(), "Namespace failed to be deleted on Remote Cluster")
+					})
+
+					It("sample app is deleted in both clusters and the namespace is empty", func(ctx SpecContext) {
+						common.CheckNamespaceEmpty(ctx, clPrimary, "sample")
+						common.CheckNamespaceEmpty(ctx, clRemote, "sample")
+						Success("Sample app is deleted in both clusters")
+					})
+				})
```

Comment:
Again unsure what this is testing. Maybe it should be an `After` step if it's just cleanup?

Reply:
Moving this also to the clean up